application/event_processor/event_processor.py

```diff
@@ -1,75 +1,198 @@
+        """Using the nhs entity attributed to this object, it finds the 
+           matching DoS services from the db and filters the results
+        """
 
         # Check database for services with same first 5 digits of ODSCode
         matching_services = get_matching_dos_services(self.nhs_entity.ODSCode)
-        logger.info(
-            f"Found {len(matching_services)} services in DB with matching first 5 chars of ODSCode: {matching_services}"
-        )
+        logger.info(f"Found {len(matching_services)} services in DB with "
+                    f"matching first 5 chars of ODSCode: {matching_services}")
 
         # Filter for services with valid type and status
-        self.matching_services = [
-            s for s in matching_services if int(s.typeid) in VALID_SERVICE_TYPES and int(s.statusid) == VALID_STATUS_ID
-        ]
-        logger.info(
-            f"Found {len(matching_services)} services with typeid in whitelist "
-            f"{VALID_SERVICE_TYPES} and status id = {VALID_STATUS_ID}: {matching_services}"
-        )
+        matching_services = [s for s in matching_services
+                             if  int(s.typeid) in VALID_SERVICE_TYPES 
+                             and int(s.statusid) == VALID_STATUS_ID]
+
+        logger.info(f"Found {len(matching_services)} services with typeid in "
+                    f"whitelist{VALID_SERVICE_TYPES} and status id = "
+                    f"{VALID_STATUS_ID}: {matching_services}")
+
+        self.matching_services = matching_services
         return self.matching_services
 
+
     def get_change_requests(self) -> dict:
-        """Generates change requests needed for the found services to make them inline with the NHS Entity
+        """Generates change requests needed for the found services to make 
+           them inline with the NHS Entity
 
         Returns:
             dict: A dictionary of change requests
         """
+        if self.matching_services is None:
+            logger.error("Attempting to form change requests before matching "
+                         "services have been found.")
+            return
+
         change_requests = []
         for service in self.matching_services:
+
             # Find changes, don't make a change req if none found
-            changes = service.get_changes(self.nhs_entity)
-            if len(changes) == 0:
-                continue
-            cr = ChangeRequest(service.id, changes).get_change_request()
-            change_requests.append(cr)
-        logger.info(f"Created {len(change_requests)} change requests {dumps(change_requests, indent=2, default=str)}")
+            changes = get_changes(service, self.nhs_entity)
+            print(f"CHANGES::: {changes}")
+            if len(changes) > 0:
+                change_requests.append(ChangeRequest(service.id, changes))
+
+        payload_list = dumps([cr.create_payload() for cr in change_requests], 
+                             indent=2, default=str)
```

Comment:
Can the indent be removed as it puts the logs across multiple lines why make it awful to track where the CR is from in Splunk

Reply:
Yep I can see that, will remove the indent.